capa/features/extractors/drakvuf/call.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2024 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Tuple, Iterator
+
+from capa.helpers import str_to_number
+from capa.features.insn import API, Number
+from capa.features.common import String, Feature
+from capa.features.address import Address
+from capa.features.extractors.base_extractor import CallHandle, ThreadHandle, ProcessHandle
+from capa.features.extractors.drakvuf.models import Call
+
+logger = logging.getLogger(__name__)
+
+
+def extract_call_features(ph: ProcessHandle, th: ThreadHandle, ch: CallHandle) -> Iterator[Tuple[Feature, Address]]:
+    """
+    This method extracts the given call's features (such as API name and arguments),
+    and returns them as API, Number, and String features.
+
+    args:
+      ph: process handle (for defining the extraction scope)
+      th: thread handle (for defining the extraction scope)
+      ch: call handle (for defining the extraction scope)
+
+    yields:
+      Feature, address; where Feature is either: API, Number, or String.
+    """
+    call: Call = ch.inner
+
+    # list similar to disassembly: arguments right-to-left, call
+    for arg_value in reversed(call.arguments.values()):
+        try:
+            yield Number(str_to_number(arg_value)), ch.address
+        except ValueError:
+            # yield argument as a string
+            yield String(arg_value), ch.address
```

Comment:
@yelhamer please comment or address

Reply:
I'm now yielding the `"ntdll.dll"` part of the argument in addition to the entire string (we yield the entire string just in case of unexpected argument formats).

Reply:
@yelhamer can you show some examples from `show-features.py`? I'm not quite following what you mean by this formatting.

Reply:
@williballenthin I meant that for `"0xc6f217efe0:\"ntdll.dll\""` for example it would yield `String("ntdll.dll")` and `String("0xc6f217efe0:\"ntdll.dll\"")`, but looking at `show-features.py` it does give misleading results (displays same argument twice):
```
yacine@y:~/src/capa/scripts$ python3 show-features.py small_drakmon.log -d
DEBUG:capa:skipping library code matching: only supported by the vivisect backend
global: global: format(pe)
global: global: os(windows)
global: global: arch(amd64)
proc: \Device\HarddiskVolume2\Windows\System32\conhost.exe (ppid=4852, pid=3564)
 proc: \Device\HarddiskVolume2\Windows\System32\conhost.exe: string(\\Device\\HarddiskVolume2\\Windows\\System32\\conhost.exe)
  thread: 6592
    call 0: LdrLoadDll(440203471832, "api-ms-win-core-fibers-l1-1-1", 0x667e2beb90:"api-ms-win-core-fibers-l1-1-1", 0, 2049)
```

With this in mind I think I might just revert to just yielding `"0xc6f217efe0:\"ntdll.dll\""` as we originally planned, since it would show up in `show-features.py` and might give analysts more insights, and it also wouldn't be misleading like yielding just `"ntdll.dll"`, and finally I don't imagine we would be missing any rule matches by yielding `"0xc6f217efe0:\"ntdll.dll\""` because the relevant api function would be expecting a memory address so I wouldn't imagine any rules basing any logic on that. Thoughts?